src/test/java/com/checkmarx/flow/cucumber/integration/end2end/genericendtoend/BugTracker.java

```diff
@@ -110,11 +109,101 @@ void deleteIssues() {
+        private  Integer projectId = null;
+        private  GitLabProperties gitLabProperties;
+        @Override
+        void init(GenericEndToEndSteps genericEndToEndSteps) {
+            gitLabProperties = genericEndToEndSteps.gitLabProperties;
+            projectName = genericEndToEndSteps.getRepository().getRepoName();
+            projectId = getProjectId();
+        }
+
+        @Override
+        void verifyIssueCreated(String severities, String engine) {
+            for (int retries = 0; retries < NUMBER_OF_RETRIES; retries++) {
+                try {
+                    TimeUnit.SECONDS.sleep(RETRY_TIMEOUT_IN_SECONDS);
+                    log.info("checking for issues in gitlab project {} starting attempt {}", projectId, retries + 1);
+                } catch (Exception e) {
+                    log.error("error in timeout while verifying created issues: {}", e.getMessage());
+                }
+                JSONArray issuesList = getAllGitlabIssues();
+                if (issuesList.length() > 0){
+                    log.info("successfully verified created issues!");
+                    return;
+                }
+            }
+            String msg = String.format("didn't find any security issues in gitlab project %d after %d retries", projectId, NUMBER_OF_RETRIES);
+            log.error(msg);
+            fail(msg);
+        }
+
+        @Override
+        void deleteIssues() {
+            JSONArray issuesArray = getAllGitlabIssues();
+            log.info("going to delete {} issues from gitlab project", issuesArray.length());
+
+            for (int i=0; i < issuesArray.length(); i++){
+                JSONObject issueObj = issuesArray.getJSONObject(i);
+                int issueIid = issueObj.getInt("iid");
+                String getProjectsUrl = String.format("%s%s", gitLabProperties.getApiUrl(), DELETE_ISSUE_URL);
+                HttpEntity<String> httpEntity = new HttpEntity<>(getHeaders());
+                ResponseEntity<String> response = restTemplate.exchange(getProjectsUrl, HttpMethod.DELETE, httpEntity, String.class, projectId, issueIid);
+
+                if (response.getStatusCode().equals(HttpStatus.NO_CONTENT)) {
+                    log.info("deleted issue iid {} successfully", issueIid);
+                }
+                else{
+                    log.warn("failed to delete issue {}. response status: {}", issueIid, response.getStatusCode());
+                }
+            }
+        }
+
+        private int getProjectId()
+        {
+            String getProjectsUrl = String.format("%s%s", gitLabProperties.getApiUrl(), GET_PROJECT_URL);
+            HttpEntity<String> httpEntity = new HttpEntity<>(getHeaders());
+            ResponseEntity<String> response = restTemplate.exchange(getProjectsUrl, HttpMethod.GET, httpEntity, String.class, projectName);
+
+            JSONArray jsonArray = new JSONArray(response.getBody());
+            JSONObject obj = jsonArray.getJSONObject(0);
```

Comment:
It's not clear what the meaning of jsonArray and obj is. It's better to make the variable names more informative.
Also please clarify why we take 0.

Reply:
fixed names